src/overlays/actors/ovl_Dm_Bal/z_dm_bal.c

```diff
@@ -30,31 +36,185 @@ const ActorInit Dm_Bal_InitVars = {
+
+    if (Cutscene_CheckActorAction(globalCtx, 0x238)) {
+        actionIndex = Cutscene_GetActorActionIndex(globalCtx, 0x238);
+
+        if (D_80C1F2C0 != globalCtx->csCtx.actorActions[actionIndex]->action) {
+            D_80C1F2C0 = globalCtx->csCtx.actorActions[actionIndex]->action;
+            switch (globalCtx->csCtx.actorActions[actionIndex]->action) {
+                case 1:
+                    this->unk336 = 0;
+                    this->unk198 = 0;
+                    Actor_ChangeAnimationByInfo(&this->skelAnime, D_80C1F170, 0);
+                    break;
+                case 2:
+                    this->unk336 = 1;
+                    Actor_ChangeAnimationByInfo(&this->skelAnime, D_80C1F170, 12);
+                    break;
+                case 3:
+                    Actor_ChangeAnimationByInfo(&this->skelAnime, D_80C1F170, 13);
+                    break;
+            }
+        } else if (D_80C1F2C0 == 3) {
+            if (Animation_OnFrame(&this->skelAnime, 0.0f)) {
+                this->unk336 = 1;
+            } else if (Animation_OnFrame(&this->skelAnime, 29.0f)) {
+                this->unk336 = 0;
+                this->unk198 = 0;
+            }
+        }
+        Cutscene_ActorTranslateAndYaw(&this->actor, globalCtx, actionIndex);
+        this->actor.home.pos = this->actor.world.pos;
+        return;
+    } else {
+        this->unk336 = 0;
+        this->unk198 = 0;
+        D_80C1F2C0 = 0x63;
+    }
+}
+
+Vec3f D_80C1F2C4 = { 0.0f, 9.0f, 0.0f }; // Data location necessary to match, as func_80C1EAE8 contains D_80C1F2C0
+u32 D_80C1F2D0[] = { 0x06006050, 0x060094D0, 0, 0 };
+
+void func_80C1EC60(DmBal* this, GlobalContext* globalCtx) {
+    f32 temp_fv1;
+    f32 temp_fv1_2;
+
+    this->unk338 += 0x320;
+    this->unk33A += 0x3E8;
+    temp_fv1 = (Math_CosS(this->unk338) * 0.1f) + 1.0f;
+    this->unk194 = temp_fv1;
+    this->unk190 = temp_fv1;
+    temp_fv1_2 = (Math_SinS(this->unk338) * 0.1f) + 1.0f;
+    this->unk18C = temp_fv1_2 * temp_fv1_2;
+    this->actor.world.pos.y = (Math_SinS(this->unk338) * 25.0f) + this->actor.home.pos.y;
+}
+
+void func_80C1ED0C(DmBal* this) {
+    s16 temp_v0;
+
+    if (this->unk336 == (s16)1) {
+        this->unk198 = 1;
+        return;
+    }
+    temp_v0 = this->unk334;
+    if (temp_v0 >= 4) {
+        this->unk334 = (s16)(temp_v0 - 1);
+        return;
+    }
+    if (temp_v0 != 0) {
+        this->unk198 = 1;
+        this->unk334 = (s16)(temp_v0 - 1);
+        return;
+    }
+    this->unk198 = 0;
+    this->unk334 = 0x3C;
+}
+
+void func_80C1ED64(DmBal* this, GlobalContext* globalCtx, Vec3f* arg2, Vec3f* arg3, f32 arg4) {
+    Actor* paper;
+
+    paper = Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_PAPER, arg2->x, arg2->y, arg2->z, 0, 0, 0, 0);
+    if (paper != NULL) {
+        paper->velocity = *arg3;
+        paper->gravity = arg4;
+    }
+}
+
+void DmBal_Update(Actor* thisx, GlobalContext* globalCtx) {
+    DmBal* this = THIS;
+    f32 sp44;
```

Comment:
```suggestion
    s32 pad;
    DmBal* this = THIS;
```
Looks like `sp44` is actually some pad variable.